java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -1036,4 +1037,100 @@ CompletableFuture<String[]> xclaimJustId(
             long minIdleTime,
             String[] ids,
             StreamClaimOptions options);
+
+    /**
+     * Returns the list of all consumer groups and their attributes for the stream stored at <code>key
+     * </code>.
+     *
+     * @see <a href="https://valkey.io/commands/xinfo-groups/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @return An <code>Array</code> of mappings, where each mapping represents the attributes of a
+     *     consumer group for the stream at <code>key</code>.
+     * @example
+     *     <pre>{@code
+     * Map[] groups = client.xinfoGroups(key).get();
+     * for (int i = 0; i < groups.length; i ++) {
+     *     System.out.println("Info of group: " + groups[0].get("name"));
+     *     System.out.println("\tname: " + groups[0].get("name"));
+     *     System.out.println("\tconsumers: " + groups[0].get("consumers"));
+     *     System.out.println("\tpending: " + groups[0].get("pending"));
+     *     System.out.println("\tlast-delivered-id: " + groups[0].get("last-delivered-id"));
+     *     System.out.println("\tentries-read: " + groups[0].get("entries-read"));
+     *     System.out.println("\tlag: " + groups[0].get("lag"));
+     * }
+     * }</pre>
+     */
+    CompletableFuture<Map<String, Object>[]> xinfoGroups(@NonNull String key);
+
+    /**
+     * Returns the list of all consumer groups and their attributes for the stream stored at <code>key
+     * </code>.
+     *
+     * @see <a href="https://valkey.io/commands/xinfo-groups/">valkey.io</a> for details.
+     * @param key The key of the stream.
+     * @return An <code>Array</code> of mappings, where each mapping represents the attributes of a
+     *     consumer group for the stream at <code>key</code>.
+     * @example
+     *     <pre>{@code
+     * Map[] groups = client.xinfoGroups(key).get();
+     * for (int i = 0; i < groups.length; i ++) {
+     *     System.out.println("Info of group: " + groups[0].get("name"));
```

Comment:
Please doulbe check map key type and update all examples if needed
```suggestion
     *     System.out.println("Info of group: " + groups[0].get(gs("name")));
```

Reply:
The return here is a string. Calling gs() returns null here

Reply:
Yes, I see your tests. So return type should be updated or there is an error in value conversion

Reply:
If you fix to return a map with `GlideString`s, please update docs and tests